test/entrypoint.test.ts

```diff
@@ -433,6 +433,69 @@ describe('EntryPoint', function () {
         await calcGasUsage(rcpt, entryPoint, beneficiaryAddress)
       })
 
+      // rimeissner:
+      it('account should pay for high gas usage tx', async function () {
+        const iterations = 45
+        const count = await counter.populateTransaction.gasWaster(iterations, '')
+        const accountExec = await account.populateTransaction.execute(counter.address, 0, count.data!)
+        const op = await fillAndSign({
+          sender: account.address,
+          callData: accountExec.data,
+          verificationGasLimit: 1e5,
+          callGasLimit: 11e5
+        }, accountOwner, entryPoint)
+        const beneficiaryAddress = createAddress()
+        const offsetBefore = await counter.offset()
+        console.log('  == offset before', offsetBefore)
+        // for estimateGas, must specify maxFeePerGas, otherwise our gas check fails
+        console.log('  == est gas=', await entryPoint.estimateGas.handleOps([op], beneficiaryAddress, { maxFeePerGas: 1e9 }).then(tostr))
+
+        // must specify at least on of maxFeePerGas, gasLimit
+        // (gasLimit, to prevent estimateGas to fail on missing maxFeePerGas, see above..)
+        const rcpt = await entryPoint.handleOps([op], beneficiaryAddress, {
+          maxFeePerGas: 1e9,
+          gasLimit: 13e5
+        }).then(async t => await t.wait())
+
+        console.log('rcpt.gasUsed=', rcpt.gasUsed.toString(), rcpt.transactionHash)
+        await calcGasUsage(rcpt, entryPoint, beneficiaryAddress)
+
+        // check that the state of the counter contract is updated
+        // this ensures that the `callGasLimit` is high enough
+        // therefore this value can be used as a reference in the test below
+        console.log('  == offset after', await counter.offset())
+        expect(await counter.offset()).to.equal(offsetBefore.add(iterations))
+      })
+
+      it('account should not pay if too low gas limit was set', async function () {
```

Comment:
I assume this test passes now with the above fix?

Reply:
yes, of course, all tests pass now (and dind't before the fix.